internal/core/services/user_device_service.go

```diff
@@ -265,6 +266,33 @@ func (das *userDeviceAPIService) CanRequestInstantOffer(ctx context.Context, use
 	return true, nil
 }
 
+func (das *userDeviceAPIService) LastRequestDidGiveError(ctx context.Context, userDeviceID string) (bool, error) {
+
+	existingOfferData, err := models.Valuations(
+		models.ValuationWhere.UserDeviceID.EQ(null.StringFrom(userDeviceID)),
+		models.ValuationWhere.OfferMetadata.IsNotNull(),
+		qm.OrderBy("updated_at desc"), qm.Limit(1)).
+		One(ctx, das.dbs().Reader)
+
+	if err != nil && !errors.Is(err, sql.ErrNoRows) {
+		return false, err
```

Comment:
if there were no rows returns, means they have never request valuation before - in this case we should return false but error nil